RealmSwift/LinkingObjects.swift

```diff
@@ -422,6 +429,39 @@ extension LinkingObjects : RealmCollection {
     }
 }
 
+private struct LinkingObjectsHandoverMetadata {
+    var propertyName: String
+    var property: RLMProperty?
```

Comment:
If I'm understanding this correctly, `property` is only relevant if the `RLMResults` instance that backs the `LinkingObjects` has yet to be populated, and in that case it must be non-nil. Using an enum rather than a struct for this metadata type may allow us to communicate that relationship more clearly.